Cython/Compiler/PyrexTypes.py

```diff
@@ -3776,32 +3776,57 @@ def can_coerce_from_pyobject(self, env):
             return True
         return False
 
+    def _get_conversion_context_cls_cname(self, cname_format_str):
+        # cname_format_str is passed "prefix", "format", "tags"
+        X = "XYZABC"
+        tags = []
+        context = {}
+        optional_tag_strs = []
+        optional_arg_template_names = ['']  # these may be different (e.g. for unordered_map, map)
+            # hence have to be generated here rather than hard-coded in the utility_code.
+            # Start with an empty string so that the joined output starts with ','
+        for ix, T in enumerate(self.templates or []):
+            if ix < builtin_cpp_conversions[self.cname]:
+                tags.append(T.specialization_name())
+                context[X[ix]] = T
+            else:
+                optional_arg_template_names.append("%s=*" % self.template_type.templates[ix].name)
+                if not isinstance(T, TemplatePlaceholderType):
+                    tags.append(T.specialization_name())
+                    optional_tag_strs.append(X[ix])
+                    context[X[ix]] = T
+
+        optional_tag_strs = ",".join(optional_tag_strs)
+        optional_arg_template_names = ",".join(optional_arg_template_names)
+
+        if self.cname in cpp_string_conversions:
+            cls = 'string'
```

Comment:
`cls` is the typical thing that you find in classmethods. Let's invest a couple of characters into clarity and name this `cpp_class`.

Reply:
Actually, `cpp_class_name`. Why not.